### ISSUE TITLE
Change queries in TestReportStream.kt to allow for running tests in parallel

### ISSUE THREAD

**Issue by jimduff-usds** [CDCgov/prime-reportstream#1961]
Replace all calls to `reportLineageCountQuery()` with calls to `itemLineageCountQuery()`.  They don't work quite the same, so you might need to rejigger things a bit.

- Please delete `reportLineageCountQuery()` as part of this.

Any test that uses reportLineageCountQuery could deliver wrong results when the tests are run in parallel.

Item lineages are cleanly immune to merging - individual items can only descend from a single originally submitted parent.
This is quite unlike reports as a whole (the REPORT_FILE table) which can contain data from many parents, so the `report_file.item_count` field data gets all messed up and confused when unwanted merging happens.

See also #1954 and related tickets.

**Comment by carlosfelix2** [comment]
Duplicate of #1960
